esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -4522,6 +4543,8 @@ void deploySharedObject(IEspContext &context, StringBuffer &wuid, const char *fi
             wu->setJobName(srcxml->queryProp("@jobName"));
         if (srcxml->hasProp("@token"))
             wu->setSecurityToken(srcxml->queryProp("@token"));
+        if (srcxml->hasProp("@signature"))
+            wu->setSecuritySignature(srcxml->queryProp("@signature"));
```

Comment:
There's a chance when copying workunit shared objects that in the end the WUID will be mangled for uniqueness, (See line 4500/4521 comment "duplicate wuid made unique"), which would change the signature.  I think it would be better to call the method that regenerates the signature here.  
But with that change I would also move this to the bottom so that signing the workunit was one of the last things that you do.  Perhaps right before wu->setState(..) below.

Reply:
Done. I do find it odd that this method dereferences wu with both wu->xxx and wu.xxx